src/pyFAI/test/test_fiber_integrator.py

```diff
@@ -0,0 +1,312 @@
+#!/usr/bin/env python
+# coding: utf-8
+#
+#    Project: Azimuthal integration
+#             https://github.com/silx-kit/pyFAI
+#
+#    Copyright (C) 2015-2024 European Synchrotron Radiation Facility, Grenoble, France
+#
+#    Principal author:       Jérôme Kieffer (Jerome.Kieffer@ESRF.eu)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+
+"""test suite for Fiber integrator class"""
+
+__author__ = "Edgar Gutiérrez Fernández"
+__contact__ = "edgar.gutierrez-fernandez@esrf.fr"
+__license__ = "MIT"
+__copyright__ = "European Synchrotron Radiation Facility, Grenoble, France"
+__date__ = "23/10/2024"
+
+import unittest
+import numpy
+import logging
+from ..calibrant import get_calibrant
+logger = logging.getLogger(__name__)
+from ..integrator.fiber import FiberIntegrator
+from ..detectors import detector_factory
+from ..units import get_unit_fiber
+
+
+class TestFiberIntegrator(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        dist = 0.1
+        poni1 = 0.02
+        poni2 = 0.02
+        detector = detector_factory("Eiger2_4M")
```

Comment:
The time spent for running test is large, keep the amount of work as small as (reasonably) possible. Is it possible to reduce the size of the detector to "pilatus100k" for example ? Each test would be 40x faster.